source/backend/dal/Helpers/Extensions/EntityExtensions.cs

```diff
@@ -135,8 +137,11 @@ public static T ThrowIfNotAllowedToEdit<T>(this T entity, string paramName, Clai
 
             foreach (var oldItem in dbItemsMap.Values)
             {
-                accessor.Remove(dbEntity, oldItem);
-                context.Remove(oldItem);
+                if (children.FirstOrDefault(x => x.Internal_Id.Equals(oldItem.Internal_Id)) == null)
```

Comment:
Right, I guess we either need this or line 133 should get moved out of the if on 130.

Reply:
yes, before this change the method was removing the entity even if was not intended due to the new flag